### PR TITLE
Resolved Issue #2939

### PR DESCRIPTION
**Pull request by avibega23**
1. Input Validation in main() function:
Issue: The original code did not check for non-positive values of n (array size). As a result, entering a value of n <= 0 would cause buffer overflow or undefined behavior when the array was accessed.

Change: Added input validation for the array size n to ensure it is positive. If n <= 0, the program prints an error message and terminates gracefully:

if (n <= 0) {
    std::cerr << "Error: Array size must be a positive integer.\n";
    return 1;
}

2. Handling Empty Arrays in median_of_medians() function:
Issue: In the original code, if the array was empty or the median vector m ended up being empty, the program would try to access m[0], causing a buffer overflow.

Change: Before accessing m[(sz - 1) / 2], a check was added to ensure that the median vector m is not empty. If the vector is empty, an error message is printed, and the program exits:

if (m.empty()) {
    std::cerr << "Error: Median vector is empty.\n";
    exit(1);
}

3. Graceful Error Handling for Invalid Inputs:
Issue: The code previously did not handle invalid inputs properly, and it would crash with a segmentation fault when invalid input was given.

Change: Instead of continuing with invalid or empty inputs, the program now handles such inputs gracefully by printing error messages and terminating cleanly, thus preventing crashes and undefined behavior.

Example: If n <= 0 is entered, the program prints:

"Error: Array size must be a positive integer."

4. Edge Case Handling in Test Cases:
Issue: The original test cases did not cover edge cases like empty arrays or non-positive sizes.

Change: Though not explicitly mentioned in the test section, handling of invalid inputs was prioritized in the main function, ensuring no test cases would be executed for invalid inputs.

### CODE REVIEW

**Comment by Bimaah** [comment]
Para de me mandar e-mails seus desgraçados

Em ter., 29 de abr. de 2025, 08:47, Avinoor Singh ***@***.***>
escreveu:

File: search/median_search.cpp [hunk 1]
```diff
@@ -131,7 +136,11 @@ int main()
 	int n = 0;
 	std::cout << "Enter Size of Array: ";
 	std::cin >> n;
-	std::vector<int> a(n);
+    if (n <= 0) {
+        std::cerr << "Error: Array size must be a positive integer.\n";
```
Comment (realstealthninja) [inline search/median_search.cpp:140]: shouldn't this throw an error instead?

**Review by realstealthninja — CHANGES_REQUESTED** [review]
